test/unit/intrinsic/src/Ffs.cpp

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/intrinsic/Traits.hpp>
+
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+#include <alpaka/test/KernelExecutionFixture.hpp>
+
+#include <catch2/catch.hpp>
+
+//#############################################################################
+template<
+    typename TInput>
+class FfsTestKernel
+{
+public:
+    //-----------------------------------------------------------------------------
+    ALPAKA_NO_HOST_ACC_WARNING
+    template<
+        typename TAcc>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const & acc,
+        bool * success) const
+    -> void
+    {
+        TInput const inputs[] = {0, 1, 3, 54, 163, 51362,
+            -43631, -1352, -642, -1};
+        for( auto const input : inputs )
+        {
+            int const expected = ffsNaive(input);
+            int const actual = alpaka::intrinsic::ffs(acc, input);
+            ALPAKA_CHECK(*success, actual == expected);
+        }
+    }
+
+private:
+    ALPAKA_FN_ACC static auto ffsNaive(TInput value) -> int
+    {
+        if (value == 0)
+            return 0;
+        int result = 1;
+        while ((value & 1) == 0)
+        {
+            value >>= 1;
+            result++;
+        }
+        return result;
+    }
+};
+
+//-----------------------------------------------------------------------------
+TEMPLATE_LIST_TEST_CASE( "ffs", "[intrinsic]", alpaka::test::acc::TestAccs)
+{
+    using Acc = TestType;
+    using Dim = alpaka::dim::Dim<Acc>;
+    using Idx = alpaka::idx::Idx<Acc>;
+
+    alpaka::test::KernelExecutionFixture<Acc> fixture(
+        alpaka::vec::Vec<Dim, Idx>::ones());
+
+    FfsTestKernel<int> kernel32bit;
+    REQUIRE(
+        fixture(
+            kernel32bit));
+
+    FfsTestKernel<long long> kernel64bit;
```

Comment:
IMO we should check `int64_t` and `int32_t`, Those types are the typical used by the user and maps to `long long` or `long` but this is OS dependent.

Reply:
That is a good point. The problem is that how GCC and CUDA define it is for these types, of not fixed width. I can make alpaka interface more modern C++ and `static_assert` internally that the type matches. Are you fine with it?

Reply:
@psychocoderHPC I guess my last answer also relates to your [earlier comment](https://github.com/alpaka-group/alpaka/pull/1018#issuecomment-635957171), not just the one above.